packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -198,40 +237,103 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
 
   @override
   Widget build(BuildContext context) {
-    final List<DataRow> rows = _getRows(_firstRowIndex, config.rowsPerPage);
-    Widget table = new DataTable(
-      key: _tableKey,
-      columns: config.columns,
-      sortColumnIndex: config.sortColumnIndex,
-      sortAscending: config.sortAscending,
-      onSelectAll: config.onSelectAll,
-      rows: rows
-    );
+    final TextStyle textStyle = Theme.of(context).textTheme.caption;
+    final List<Widget> footerWidgets = <Widget>[];
+    if (config.onRowsPerPageChanged != null) {
+      footerWidgets.addAll(<Widget>[
+        new Text(
+          'Rows per page:'
+        ),
+        new DropDownButtonHideUnderline(
+          child: new DropDownButton<int>(
+            items: config.availableRowsPerPage
+                     .where((int value) => value <= _rowCount)
+                     .map/*<DropDownMenuItem<int>>*/((int value) {
+                       return new DropDownMenuItem<int>(
+                         value: value,
+                         child: new Text('$value')
+                       );
+                     }).toList(),
+            value: config.rowsPerPage,
+            onChanged: config.onRowsPerPageChanged,
+            style: textStyle,
+            iconSize: 24.0
+          )
+        ),
+      ]);
+    }
+    footerWidgets.addAll(<Widget>[
+      new Container(width: 32.0),
+      new Text(
+        '${_firstRowIndex + 1}\u2013${_firstRowIndex + config.rowsPerPage} ${ _rowCountApproximate ? "of about" : "of" } $_rowCount'
+      ),
+      new Container(width: 32.0),
+      new IconButton(
+        padding: new EdgeInsets.all(0.0),
+        icon: Icons.chevron_left,
+        onPressed: _firstRowIndex <= 0 ? null : () {
+          pageTo(math.max(_firstRowIndex - config.rowsPerPage, 0));
+        }
+      ),
+      new Container(width: 24.0),
+      new IconButton(
+        padding: new EdgeInsets.all(0.0),
+        icon: Icons.chevron_right,
+        onPressed: (!_rowCountApproximate && (_firstRowIndex + config.rowsPerPage >= _rowCount)) ? null : () {
+          pageTo(_firstRowIndex + config.rowsPerPage);
+        }
+      ),
+      new Container(width: 14.0),
+    ]);
     return new Card(
       // TODO(ianh): data table card headers
-      child: table
-      // TODO(ianh): data table card footers: prev/next page, rows per page, etc
+      /*
+         - title, top left
+            - 20px Roboto Regular, black87
+         - persistent actions, top left
+         - header when there's a selection
+            - accent 50?
+            - show number of selected items
+            - different actions
+         - actions, top right
+            - 24px icons, black54
+      */
+      child: new BlockBody(
+        children: <Widget>[
+          new IntrinsicHeight(
+            // TODO(ianh): find a better way to do scrolling of one-child viewports than Block
+            // (that would let us get rid of the IntrinsicHeight widget, at least)
```

Comment:
nothing's wrong with it, I just didn't think about it. I've added some pointers to it from Block.